src/main/java/org/kuali/kfs/sys/document/service/impl/AccountingLineRuleHelperServiceImpl.java

```diff
@@ -0,0 +1,589 @@
+    }
+
+    @Override
+    public String getObjectCodeLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(ObjectCode.class.getName())
+                .getAttributeDefinition(KFSPropertyConstants.FINANCIAL_OBJECT_CODE).getShortLabel();
+    }
+
+    @Override
+    public String getObjectSubTypeCodeLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(ObjectSubType.class.getName())
+                .getAttributeDefinition(KFSPropertyConstants.CODE).getShortLabel();
+    }
+
+    @Override
+    public String getObjectTypeCodeLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(ObjectType.class.getName())
+                .getAttributeDefinition(KFSConstants.GENERIC_CODE_PROPERTY_NAME).getShortLabel();
+    }
+
+    @Override
+    public String getOrganizationCodeLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(Organization.class.getName())
+                .getAttributeDefinition(KFSPropertyConstants.ORGANIZATION_CODE).getShortLabel();
+    }
+
+    @Override
+    public String getProjectCodeLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(ProjectCode.class.getName())
+                .getAttributeDefinition(KFSPropertyConstants.CODE).getShortLabel();
+    }
+
+    @Override
+    public String getSubAccountLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(SubAccount.class.getName())
+                .getAttributeDefinition(KFSPropertyConstants.SUB_ACCOUNT_NUMBER).getShortLabel();
+    }
+
+    @Override
+    public String getSubFundGroupCodeLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(SubFundGroup.class.getName())
+                .getAttributeDefinition(KFSPropertyConstants.SUB_FUND_GROUP_CODE).getShortLabel();
+    }
+
+    @Override
+    public String getSubObjectCodeLabel() {
+        return businessObjectDictionaryService.getBusinessObjectEntry(SubObjectCode.class.getName())
+                .getAttributeDefinition(KFSPropertyConstants.FINANCIAL_SUB_OBJECT_CODE).getShortLabel();
+    }
+
+    @Override
+    public boolean hasRequiredOverrides(AccountingLine line, String overrideCode) {
+        return hasAccountRequiredOverrides(line, overrideCode) && hasObjectBudgetRequiredOverrides(line, overrideCode);
+    }
+
+    public boolean hasAccountRequiredOverrides(AccountingLine line, String overrideCode) {
+        boolean retVal = true;
+        AccountingLineOverride override = AccountingLineOverride.valueOf(overrideCode);
+        Account account = line.getAccount();
+        if (AccountingLineOverride.needsExpiredAccountOverride(account)
+                && !override.hasComponent(AccountingLineOverride.COMPONENT.EXPIRED_ACCOUNT)) {
+            Account continuation = accountService.getUnexpiredContinuationAccountOrNull(account);
+            if (continuation == null) {
```

Comment:
This spot wasn't impacted by the FINP-7183 changes, but it may still be a good idea to perform the ObjectUtils null check here.

Reply:
Added null check.